src/NonceTracker.ts

```diff
@@ -0,0 +1,251 @@
+import assert from 'assert';
+import { Mutex } from 'await-semaphore';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
+const EthQuery = require('ethjs-query');
+
+/**
+ *  @property opts.provider - An ethereum provider
+ *  @property opts.blockTracker - An instance of eth-block-tracker
+ *  @property opts.getPendingTransactions - A function that returns an array of txMeta
+ *  whose status is `submitted`
+ *  @property opts.getConfirmedTransactions - A function that returns an array of txMeta
+ *  whose status is `confirmed`
+ */
+export interface NonceTrackerOptions {
+  provider: Record<string, unknown>;
+  blockTracker: any;
```

Comment:
I think both of these should be `any`. Generally, we always prefer `unknown` over `any` but in the case of untyped dependencies there aren't really any good alternatives.

The type of `provider` doesn't really matter here due to our usage, but `any` will make it consistent with `blockTracker`.